src/articles/2024-01-21_DDEV-for-js.md

```diff
@@ -0,0 +1,327 @@
+
+Configuration complete. You may now run 'ddev start'.
+```
+
+First, DDEV will ask for your project name. This value will be used in the primary URL for the project, so make sure it follows the rules for [hostname syntax](https://en.wikipedia.org/wiki/Hostname#Syntax). Next, DDEV will attempt to identify what kind of project you have. Since DDEV was originally built with php-based CMSes in mind, the default project type is "php", meaning there's no special extra configuration happening. There is currently [an open issue](https://github.com/ddev/ddev/issues/5722) to consider renaming this to something more helpful like "generic" or "any".
+
+If you run `ddev start` now, you'll have a working `web` and `db` container!
+
+```txt
+$ ddev start
+
+Starting my-project...
+Network ddev-my-project_default created
+Building project images...
+Project images built in 0s.
+ Container ddev-my-project-db  Created
+ Container ddev-my-project-web  Created
+ Container ddev-my-project-web  Started
+ Container ddev-my-project-db  Started
+You have Mutagen enabled and your 'php' project type doesn't have `upload_dirs` set.
+For faster startup and less disk usage, set upload_dirs to where your user-generated files are stored.
+If this is intended you can disable this warning with `ddev config --disable-upload-dirs-warning`.
+Starting Mutagen sync process...
+Mutagen sync flush completed in 1s.
+For details on sync status 'ddev mutagen st my-project -l'
+Waiting for web/db containers to become ready: [web db]
+^[Starting ddev-router if necessary...
+ Container ddev-router  Running
+Waiting for additional project containers to become ready...
+All project containers are now ready.
+Successfully started my-project
+Project can be reached at https://my-project.ddev.site https://127.0.0.1:33125
+```
+
+## Configuring for non-PHP projects
+
+While our project is running, there's some configuration we need to change. Open `my-project/.ddev/config.yml`:
+
+```yml
+name: my-project
+type: php
+docroot: ""
+php_version: "8.1"
+webserver_type: nginx-fpm
+xdebug_enabled: false
+additional_hostnames: []
+additional_fqdns: []
+database:
+  type: mariadb
+  version: "10.4"
+use_dns_when_possible: true
+composer_version: "2"
+web_environment: []
+```
+
+### Add additional hostnames
+
+DDEV is _highly_ configurable. All the key/value options are [documented here](https://ddev.readthedocs.io/en/latest/users/configuration/config). There are two main things we want to edit here. We're first going to add additional hostnames. Since keystone will be running our backend, we'll have it accessible at `api.my-project.ddev.site`, and since Sveltekit is running our frontend app, we'll have itat `app.my-project.ddev.site`. The default URL will also exist at `my-project.ddev.site`, but DDEV's container healthcheck needs to be able to access php on that URL, so we'll leave it alone.
```

Comment:
```suggestion
DDEV is _highly_ configurable. All the key/value options are [documented here](https://ddev.readthedocs.io/en/latest/users/configuration/config). There are two main things we want to edit here. We're first going to add additional hostnames. Since Keystone will be running our backend, we'll have it accessible at `api.my-project.ddev.site`, and since Sveltekit is running our frontend app, we'll have it at `app.my-project.ddev.site`. The default URL will also exist at `my-project.ddev.site`, but DDEV's container healthcheck needs to be able to access php on that URL, so we'll leave it alone.
```